plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts

```diff
@@ -574,25 +411,104 @@ export async function readMicrosoftGraphOrg(
       transformer: options.userTransformer,
       logger: options.logger,
     });
-    users.push(...usersWithFilter);
+    users = usersWithFilter;
+  } else {
+    options.logger.info(
+      'Not loading users as neither user or userGroupMember configuration has been provided',
+    );
+    users = [];
   }
-  const { groups, rootGroup, groupMember, groupMemberOf } =
-    await readMicrosoftGraphGroups(client, tenantId, {
+
+  let groups: GroupEntity[];
+  if (options.groupSearch || options.groupFilter) {
+    const {
+      groups: groups2,
+      rootGroup,
+      groupMember,
+      groupMemberOf,
+    } = await readMicrosoftGraphGroups(client, tenantId, {
       queryMode: options.queryMode,
       groupSearch: options.groupSearch,
       groupFilter: options.groupFilter,
       groupSelect: options.groupSelect,
       groupTransformer: options.groupTransformer,
       organizationTransformer: options.organizationTransformer,
+      logger: options.logger,
     });
 
-  resolveRelations(rootGroup, groups, users, groupMember, groupMemberOf);
+    groups = groups2;
```

Comment:
I'm a bit of a novice to node, hoping there is a cleaner way to do this.  Not sure how to get hte deconstruction syntax to deconstruct to an existing variable.